src/Core/Checkout/Promotion/Cart/Extension/LockExtension.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Promotion\Cart\Extension;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Struct\Struct;
+use Symfony\Component\Lock\LockInterface;
+
+/**
+ * @codeCoverageIgnore
+ */
+#[Package('checkout')]
+class LockExtension extends Struct
+{
+    /**
+     * this is the key that should be
+     * used for the cart extension
+     */
+    final public const KEY = 'promotion-cart-locks';
+
+    /**
+     * @param array<string, LockInterface> $locks
+     */
+    public function __construct(
+        private readonly array $locks
+    ) {
+    }
+
+    /**
+     * @return array<string, LockInterface>
+     */
+    public function getLocks(): array
+    {
+        return $this->locks;
+    }
```

Comment:
I would use `public readonly` for this kind of property
```suggestion
    public function __construct(
        public readonly array $locks
    ) {
    }
```